### PR TITLE
feat: allow base.defined to be overwritten

### PR DESCRIPTION
**Pull request by msuperina**
All base files have been refactored to use a base constructor which adds getters and setters based on the properties passed to it

### CODE REVIEW

File: src/base.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { scaleIdentity } from 'd3-scale';
```
Comment (ColinEberhardt) [inline src/base.js]: unused import

File: src/base.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { scaleIdentity } from 'd3-scale';
+
+export default initialValues => {
+
+    const env = Object.assign({}, initialValues);
```
Comment (ColinEberhardt) [inline src/base.js]: Object.assign and Object.keys are not available in pre-Edge IE. I'm happy for us to use them, but we do need to mention that in the docs elsewhere:

https://github.com/d3fc/d3fc/blob/master/site/src/introduction/getting-started.md

> d3fc supports modern browsers. For pre-Edge versions of IE, you'll need a couple of polyfills in order to run d3fc code within the browser. The first is a polyfill for Symbol, such as es6-symbol, or the general babel-polyfill, and the second is a polyfill for CustomEvent, such as custom-event-polyfill.

After this PR is merged, can you update the above to mention the need for an additional polyfill?

Comment (msuperina) [reply src/base.js]: Apologies, I am so used to `Object.assign` and `Object.keys` I forgot they are still not supported somewhere. I will update the `d3fc` docs as well.

**Comment by msuperina** [comment]
@ColinEberhardt I added a necessary fix in commit 6567dd9 as I realized `xScale` and `yScale` were not overwritten anymore. Let me know if the `createBase` function is still a good idea as we now need to consume `xScale` and `yScale` as methods of `base`.

**Pull request merged**